protostar/cli/command_names_delayed_provider.py

```diff
@@ -0,0 +1,18 @@
+from typing import Optional
+
+from protostar.configuration_file import CommandNamesProvider
+
+
+class CommandNamesDelayedProvider(CommandNamesProvider):
```

Comment:
Can you please explain the purpose of this class?

Reply:
To transform ConfigurationFile V1 into V2, command names must be known, and they must be the same as command names provided to ProtostarCLI.

This class also breaks the dependency cycle:
`ConfigurationFile <- Commands <- ProtostarCLI <- ConfigurationFIle`